src/mbgl/util/thread.hpp

```diff
@@ -87,6 +87,34 @@ class Thread {
     RunLoop* loop = nullptr;
 };
 
+inline std::string getCurrentThreadName() {
+#if defined(__APPLE__)
+    char name[32];
+    pthread_getname_np(pthread_self(), name, sizeof(name));
+    return name;
+#elif defined(__GLIBC__) && defined(__GLIBC_PREREQ)
+#if __GLIBC_PREREQ(2, 12)
```

Comment:
@jfirebaugh I moved `name` out of the ifdefs so it'll return `"unknown"` by default.